pkg/controller/sidecar/sync.go

```diff
@@ -106,33 +107,71 @@ func (c *Controller) decommissionNode(ctx context.Context, svc *corev1.Service)
 	return nil
 }
 
-func (c *Controller) syncHostIDAnnotation(ctx context.Context, svc *corev1.Service) error {
+func (c *Controller) syncAnnotations(ctx context.Context, svc *corev1.Service) error {
 	startTime := time.Now()
-	klog.V(4).InfoS("Started syncing HostID annotation", "Service", klog.KObj(svc), "startTime", startTime)
+	klog.V(4).InfoS("Started syncing Service annotation", "Service", klog.KObj(svc), "startTime", startTime)
 	defer func() {
-		klog.V(4).InfoS("Finished syncing HostID annotation", "Service", klog.KObj(svc), "duration", time.Since(startTime))
+		klog.V(4).InfoS("Finished syncing Service annotation", "Service", klog.KObj(svc), "duration", time.Since(startTime))
 	}()
 
-	hostID, err := c.getHostID(ctx)
+	scyllaClient, err := controllerhelpers.NewScyllaClientForLocalhost()
+	if err != nil {
+		return fmt.Errorf("can't create a new ScyllaClient for localhost: %w", err)
+	}
+	defer scyllaClient.Close()
+
+	hostID, err := c.getHostID(ctx, scyllaClient)
 	if err != nil {
 		return fmt.Errorf("can't get HostID: %w", err)
 	}
 
-	hostIDValue, hasHostIDAnnotation := svc.Annotations[naming.HostIDAnnotation]
-	klog.V(4).InfoS("Syncing HostID annotation", "Service", klog.KObj(svc), "required", hostID, "existing", hostIDValue)
-	if hasHostIDAnnotation && hostIDValue == hostID {
-		klog.V(4).InfoS("Existing HostID matches the required one. Skipping update.", "Service", klog.KObj(svc))
-		return nil
+	ipToHostIDMap, err := scyllaClient.GetIPToHostIDMap(ctx, localhost)
+	if err != nil {
+		return fmt.Errorf("can't get host id to ip mapping: %w", err)
+	}
+
+	var localIP string
+	for ip, id := range ipToHostIDMap {
+		if id == hostID {
+			localIP = ip
+			break
+		}
+	}
+
+	if len(localIP) == 0 {
+		return fmt.Errorf("local host ID %q not found in IP to hostID mapping: %v", hostID, ipToHostIDMap)
+	}
+
+	nodeTokens, err := scyllaClient.GetNodeTokens(ctx, localhost, localIP)
```

Comment:
if the node IP changes - is the automatic replacement synchronous (before ScyllaDB starts serving its API) or GetIPToHostIDMap and GetNodeTokens are gonna race over the change?

Reply:
kind of wish this wasn't over IP but hostid

Reply:
API requires IP address, mapping from hostid to IP is also taken from API, so it shouldn't change in between calls, as restart is required to change Scylla config.